dod-advana/gamechanger-web
# Task/UOT-137615: code formatter setup + format existing files

## Description

1. Added Prettier to the frontend/backend as an auto-formatter
2. Configured Eslint to treat Prettier as a ruleset (resulting in Eslint errors based on required Prettier formatting)
3. Created a [Confluence page](https://confluence.di2e.net/display/UOT/GC+Web+Auto-Formatter+Configuration) with setup instructions/general info
4. Formatted all javascript files in project to get them up to date

## !vibez

phew

## Related Issue/Ticket




## Types of changes

## Smoke testing instructions

1. If you haven't, please setup Eslint with VSCode by following [instructions here](https://confluence.di2e.net/display/UOT/GC+Web+Eslint+Configuration).
2. Follow instructions on [Confluence page](https://confluence.di2e.net/display/UOT/GC+Web+Auto-Formatter+Configuration) for local Prettier setup + VSCode integration
3. Write some ugly JS code (use double quotes, incorrect indentation, no semi colons, etc.)
4. Verify that the ugly JS code shows little red squiggles in VSCode (Ie, is throwing an Eslint error that says prettier as the source):
![Screen Shot 2022-03-30 at 12 59 44 PM](https://user-images.githubusercontent.com/98841329/160920847-98d5b47c-3c55-421d-8229-4eb87f874cff.png)
5. Verify that if you save the file it will fix the error.
6. Run $ npx prettier --check "src/**/*.js" from the frontend folder to see if there are any prettier issues I missed
7. Run $ npx prettier --write "node_app/**/*.js" from the backend folder to see if I missed anything there

## Checklist:
